azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -202,7 +217,26 @@ def should_report(is_healthy, error_state, last_timestamp, period):
         return datetime.datetime.utcnow() >= (last_timestamp + period)
 
     def put_vm_log(self, content):
-        raise NotImplementedError("Unimplemented")
+        """
+        Try to upload VM logs, a compressed zip file, via the host plugin /vmAgentLog channel.
+        :param content: the binary content of the zip file to upload
```

Comment:
wow, so the host ga plugin requires to load the whole file to memory? what is the max size of the zip, again?

Reply:
I haven't figured out a way to not load it. Everything we send using `restutil` loads the object in-memory. The restriction on the agent side for the uncompressed archive is 150 MB (from my testing on actual log data, this will result in a compressed archive of 13-15 MB). On the host, the restriction is 100 MB.

Reply:
In theory, it could. However, from the memory point of view, all this is adding is an overhead of ~15 MB (worst case scenario) every hour. Disk I/O could be a bigger problem, but we don't have any plans for limiting that in the near future.